app/Http/Controllers/CaasController.php

```diff
@@ -3,70 +3,168 @@
 namespace App\Http\Controllers;
 
 use App\Models\Announcecheck;
+use App\Models\Assistantcontact;
 use App\Models\Datacaas;
 use App\Models\Grouprole;
 use App\Models\Message;
 use App\Models\Plotting;
 use App\Models\Roles;
 use App\Models\Shift;
 use App\Models\Status;
+use Carbon\Carbon;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
 
 class CaasController extends Controller
 {
     public function index()
     {
-        $profile = Datacaas::all();
+        $path = Auth::guard('caas')->user()->role->roles->character_photo;
         $data = [
-            'title' => "Caas Dashboard",
-            'profile' => $profile,
+            'title' => "Explorer's Dashboard",
+            'path' => $path
         ];
-        return view('caas.dashboard', $data);
+        return view('caas.index', $data);
     }
-    public function checkStatus()
+    public function announcement()
     {
-        $status = Status::first();
-        return view('status.dashboard', ['status' => $status]);
+        $msg = Message::first();
+        $data = [
+            'title' => "Pengumumman",
+            'msg' => $msg
+        ];
+        return view('caas.announcement', $data);
     }
+
+    public function assistant()
+    {
+        $ass = Assistantcontact::all();
+        $data = [
+            'title' => "Kontak Asisten",
+            'assistant' => $ass
+        ];
+        return view('caas.assistant', $data);
+    }
+
+    //Schedule functions
     public function viewSchedule()
     {
-        $shifts = Shift::all();
-        return view('shift.caas', compact('shifts'));
+        if (Auth::guard('caas')->user()->status->isPass == 0 || Auth::guard('caas')->user()->announcecheck()->isPlotActive == 0) {
+            return redirect()->route('caas.dashboard');
+        } elseif (boolval(Auth::guard('caas')->user()->plots) == 1) {
+            return redirect()->route('caas.fix.schedule');
+        }
```

Comment:
```suggestion
        if (!Auth::guard('caas')->user()->status->isPass || !Auth::guard('caas')->user()->announcecheck()->isPlotActive) {
            return redirect()->route('caas.dashboard');
        } elseif (boolval(Auth::guard('caas')->user()->plots)) {
            return redirect()->route('caas.fix.schedule');
        }
```

its better to write it this way for idiomatic reason, otherwise it would be awkward to do comparison just to make it boolean when 0 already means false and >=1 already means true in boolean world

Reply:
noted bang, so i can just commit the suggestion bang?

Reply:
sure you can do that too, but please check the other one as well that looks like pretty much the same thing